src/js/tech/tech.js

```diff
@@ -764,6 +764,22 @@ class Tech extends Component {
     return {};
   }
 
+  /**
+   * Create a floating video window always on top of other windows so that users may
+   * continue consuming media while they interact with other content sites, or
+   * applications on their device.
+   *
+   * @see [Spec]{@link https://wicg.github.io/picture-in-picture}
+   *
+   * @return {Promise}
+   *         A promise with a Picture-in-Picture window.
+   *
+   * @abstract
+   */
+  requestPictureInPicture() {
+    return Promise.reject();
```

Comment:
I just realized for this, we should make sure that `Promise` is available or use `this.options_.Promise`
Like in the `play()` method: https://github.com/videojs/video.js/blob/ef49a9be7e413915961bb9ab6b20d9f5c92a907c/src/js/player.js#L2206-L2216
We support browsers that may not have Promises and we want to use the Promise polyfill that was passed to us if one is given but we also shouldn't blow up if promises aren't available. Probably should just return null in that case.

Reply:
@gkatsev `this.options_.Promise` doesn't exist in `tech.js`, it exists in `player.js`. Is that expected?

Reply:
Oh, yeah, apparently it doesn't get passed in. I guess one way of fixing it is to return null and inside of the player, see whether we got back a promise or or not and if not, return a rejected promise.
But probably easier and makes more sense for longer term to pass the promise into the tech. It can be added to the techOptions inside of `loadTech_` in player.js: https://github.com/videojs/video.js/blob/ef49a9be7e413915961bb9ab6b20d9f5c92a907c/src/js/player.js#L1067-L1083 and then you'll be able to use it like outline above.

Reply:
Fixed in https://github.com/videojs/video.js/pull/6001/commits/efe840d93c384315d975c74043c9e70b14cac7c1
Thanks!